Command/CleanMediaCommand.php

```diff
@@ -0,0 +1,134 @@
+        $finder = Finder::create();
+        $filesystem    = new Filesystem();
+        $baseDirectory = $this->getContainer()->get('sonata.media.adapter.filesystem.local')->getDirectory();
+
+        $output->writeln(sprintf('<info>Scanning upload directory: %s</info>', $baseDirectory));
+
+        foreach ($pool->getContexts() as $contextName => $context) {
+            if (!$filesystem->exists($baseDirectory.'/'.$contextName)) {
+                $output->writeln(sprintf("<info>'%s' does not exist</info>", $baseDirectory.'/'.$contextName));
+                continue;
+            }
+
+            $output->writeln(sprintf('<info>Context: %s</info>', $contextName));
+
+            $files = $finder->files()->in($baseDirectory.'/'.$contextName);
+
+            foreach ($files as $file) {
+                $filename = $file->getFilename();
+
+                if (!$this->mediaExists($filename, $contextName)) {
+                    if ($dryRun) {
+                        $output->writeln(sprintf("<info>'%s' is orphanend</info>", $filename));
+                    } else {
+                        try {
+                            $filesystem->remove($file->getRealPath());
+                            $output->writeln(sprintf("<info>'%s' was successfully removed</info>", $filename));
+                        } catch (IOException $ioe) {
+                            $output->writeln(sprintf('<error>%s</error>', $ioe->getMessage()));
+                        }
+                    }
+                } elseif ($verbose) {
+                    $output->writeln(sprintf("'%s' found", $filename));
+                }
+            }
+        }
+
+        $output->writeln('<info>done!</info>');
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getProviders()
+    {
+        if (!$this->providers) {
+            $this->providers = array();
+
+            $pool = $this->getContainer()->get('sonata.media.pool');
+
+            foreach ($pool->getProviders() as $provider) {
+                if ($provider instanceof FileProvider) {
+                    $this->providers[] = $provider->getName();
+                }
+            }
+        }
+
+        return $this->providers;
+    }
+
+    /**
+     * @param $filename
+     * @param $context
+     *
+     * @return bool
+     */
+    private function mediaExists($filename, $context = null)
+    {
+        $mediaManager = $this->getContainer()->get('sonata.media.manager.media');
+
+        $fileParts = explode('_', $filename);
+
+        if (count($fileParts) > 1 && $fileParts[0] == 'thumb') {
+            return $mediaManager->findOneBy(array(
+                    'id'      => $fileParts[1],
+                    'context' => $context,
+                )) != null;
+        }
+
+        return count($mediaManager->findBy(array(
+                'providerReference' => $filename,
+                'providers'         => $this->getProviders(),
```

Comment:
Should be `                'providerName'         => $this->getProviders(),` instead?
In my case (a project using Doctrine ORM), the command works only with the latter.